exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -22,10 +22,10 @@ type boundedMemoryQueue[T any] struct {
 
 // NewBoundedMemoryQueue constructs the new queue of specified capacity, and with an optional
 // callback for dropped items (e.g. useful to emit metrics).
-func NewBoundedMemoryQueue[T any](sizer Sizer[T], capacity int) Queue[T] {
+func NewBoundedMemoryQueue[T any](set QueueCreateSettings[T]) Queue[T] {
```

Comment:
The CreateSettings change can be extracted in a separate PR since we need that anyway.

Reply:
Submitted https://github.com/open-telemetry/opentelemetry-collector/pull/9197